src/core/services/DiscordChatEngineService.ts

```diff
@@ -213,20 +218,26 @@ export class DiscordChatEngineService implements IChatEngineService {
       message.actions.forEach(async (action) => {
         const suspectUserId = response.user.id
         const owner = response.member?.user.id
-        const isInteractible = action.onlyOwnerInteraction ? suspectUserId == owner : true;
+        const isInteractible = action.onlyOwnerInteraction ? suspectUserId == owner : true
 
         if (response.customId == action.name && isInteractible) {
           // TODO: Pass context to callback for further communication
-          action.callback(origin)
+          action.callback({ dependency: this.dependency, context: origin })
           await origin.editReply({ components: [] })
         }
       })
 
     } catch (e) {
       this.logger.error('Pending response exception:')
-      this.logger.fatal(`Exception: ${e}`);
 
-      await origin.editReply({ components: [], content:"timeout..."})
+      //TODO: VERIFY IF SENDING AN EXCEPTION ACROSS ALL ACTIONS IS NEEDED
```

Comment:
this is a concern and it might not work (actually works but this case has to be more analyzed), it would be better to have the exception at Command level but the current code is not open to modify this.